x/oracle/keeper/prophecy.go

```diff
@@ -0,0 +1,97 @@
+package keeper
+
+import (
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/Sifchain/sifnode/x/oracle/types"
+)
+
+// GetProphecies returns all prophecies
+func (k Keeper) GetProphecies(ctx sdk.Context) []types.Prophecy {
+	var prophecies []types.Prophecy
+	store := ctx.KVStore(k.storeKey)
+	iter := sdk.KVStorePrefixIterator(store, types.ProphecyPrefix)
+	for ; iter.Valid(); iter.Next() {
+		var prophecy types.Prophecy
+		k.cdc.MustUnmarshalBinaryBare(iter.Value(), &prophecy)
+		prophecies = append(prophecies, prophecy)
+	}
+	return prophecies
+}
+
+// GetProphecy gets the entire prophecy data struct for a given id
+func (k Keeper) GetProphecy(ctx sdk.Context, prophecyID []byte) (types.Prophecy, bool) {
+
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(append(types.ProphecyPrefix, prophecyID[:]...))
```

Comment:
Why do we have to do appending to get the array of prophecies stored?

Reply:
it is to get the key of a prophecy, key includes the prophecy prefix and the prophecy id.